Makefile

```diff
@@ -60,3 +60,43 @@ load-test-server-local:
 .PHONY: echo-var
 echo-var:
 	@echo "$($(VAR))"
+
+
+## Location to install dependencies to
+TOOLBIN ?= $(shell pwd)/hack/tools
+$(TOOLBIN):
+	mkdir -p $(TOOLBIN)
+
+## Tool Binaries
+CONTROLLER_GEN ?= $(TOOLBIN)/controller-gen
+CONTROLLER_TOOLS_VERSION ?= v0.19.0
+CONTROLLER_GEN_VERSION ?= $(CONTROLLER_GEN)-$(CONTROLLER_TOOLS_VERSION)
+
+.PHONY: controller-gen
+controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessary.
+$(CONTROLLER_GEN): $(TOOLBIN)
```

Comment:
Wouldn't a more appropriate target here be `$(TOOLBIN)/$(CONTROLLER_GEN_VERSION)` (with corresponding change to the above line)?